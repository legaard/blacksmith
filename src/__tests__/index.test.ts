import * as index from '../index';

describe('index', () => {
    test('should export things correctly', () => {
        // Arrange, act and assert
        expect(index).toMatchSnapshot();
    });

    test('should configure exported Fixture', () => {
        // Arrange
        const { Fixture } = index;
        const sut = new Fixture();

        // Act and assert
        expect(sut).toMatchSnapshot();
    });
});
