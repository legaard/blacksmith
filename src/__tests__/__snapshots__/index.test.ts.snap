// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should configure exported Fixture 1`] = `
Fixture {
  "_customizations": Customization {
    "_builders": Object {
      "boolean": BooleanBuilder {
        "_generator": NumberGenerator {
          "_max": 1,
          "_min": 0,
        },
        "type": "boolean",
      },
      "null": NullBuilder {
        "type": "null",
      },
      "number": NumberBuilder {
        "_generator": NumberGenerator {
          "_max": 250,
          "_min": 1,
        },
        "type": "number",
      },
      "string": StringBuilder {
        "_generator": StringGenerator {
          "_prefix": "",
        },
        "type": "string",
      },
      "symbol": SymbolBuilder {
        "_generator": StringGenerator {
          "_prefix": "",
        },
        "type": "symbol",
      },
      "undefined": UndefinedBuilder {
        "type": "undefined",
      },
    },
    "_typeAliases": Object {},
  },
  "_frozenTypes": Object {},
  "_generator": NumberGenerator {
    "_max": 75,
    "_min": 5,
  },
}
`;

exports[`index should export things correctly 1`] = `
Object {
  "Builder": [Function],
  "Customization": [Function],
  "Fixture": [Function],
  "NumberGenerator": [Function],
  "PrimitiveType": Object {
    "boolean": "boolean",
    "null": "null",
    "number": "number",
    "string": "string",
    "symbol": "symbol",
    "undefined": "undefined",
  },
  "StringGenerator": [Function],
  "createAsyncInjector": [Function],
  "createInjector": [Function],
}
`;
